### PR TITLE
735138 [ADO][ALM] Extensions tab not displayed if Error tests' status

### PR DESCRIPTION
**Pull request by dorin7bogdan**

### CODE REVIEW

File: PSModule/AbstractLauncherTaskCmdlet.cs [hunk 1]
```diff
@@ -451,6 +410,47 @@ protected virtual bool CollateResults(string resultFile, string log, string resd
...
+
+            //read result xml file
+            xmlResults = File.ReadAllText(resultFile);
+
+            if (xmlResults.IsNullOrWhiteSpace())
+            {
+                WriteDebug("Empty results file");
+                return false;
+            }
+            else
```
Comment (soleverxu) [inline PSModule/AbstractLauncherTaskCmdlet.cs:430]: this `else` clause can be omitted as the `if` block always returns `false`.

File: PSModule/AbstractLauncherTaskCmdlet.cs [hunk 2]
```diff
@@ -451,6 +410,47 @@ protected virtual bool CollateResults(string resultFile, string log, string resd
...
+            {
+                try
+                {
+                    var doc = XDocument.Parse(xmlResults);
+                    if (doc?.Root == null || !doc.Root.HasElements)
```
Comment (soleverxu) [inline PSModule/AbstractLauncherTaskCmdlet.cs:435]: Shall we also check the **root** element? For example, the name of the root element.

**Review by soleverxu — COMMENTED** [review]
See my comments inline

File: PSModule/AbstractLauncherTaskCmdlet.cs (hunk 2 above)
Comment (dorin7bogdan) [reply PSModule/AbstractLauncherTaskCmdlet.cs:435]: At least at this step, we only need to know if any xml results exist.
Eventually when the xml content will be interpreted we can check it deeper.
Currently we didn't find a test scenario where HpToolsLauncher would return an xml with invalid root name.

File: PSModule/AbstractLauncherTaskCmdlet.cs (hunk 1 above)
Comment (dorin7bogdan) [reply PSModule/AbstractLauncherTaskCmdlet.cs:430]: Yes, the `else` clause itself is not functionally important here, just the code from its body.

**Review by soleverxu — APPROVED** [review]

**Pull request merged**